### ISSUE TITLE
False positive for SA1008 when pattern matching

### ISSUE THREAD

**Issue by mheyman** [DotNetAnalyzers/StyleCopAnalyzers#3400]
Code like
```csharp
if (shape is not (Polygon or MultiPolygon))
{
}
```
causes an SA1008 "Opening parenthesis should be spaced correctly" warning when it _probably_ shouldn't.

**Comment by sharwell** [comment]
Duplicate of #3230
